app/assets/javascripts/shared/new_shareable_form.js.erb

```diff
@@ -1,5 +1,11 @@
 <% template = File.join(Rails.root, 'app/views/social_networking/shared/_sharing_fields.html') %>
 <% js_helper = Class.new.extend(ActionView::Helpers::JavaScriptHelper) %>
+<% if File.exist?(template) %>
+  <% file = js_helper.j(Erubis::Eruby.new(File.read(template)).result) %>
+<% else %>
+  <% file = "" %>
+  console.warn('Warning, app/views/social_networking/shared/_sharing_fields.html is not defined in host app and sharing is not possible.');
```

Comment:
What do you think about adding logic to a Ruby initializer instead? This warning depends on someone opening up a particular page, whereas if you checked for the file on initialization a developer would be notified immediately.